server/src/parse-excel-file.ts

```
@@ -0,0 +1,30 @@
+import { assertTimescaleDataWithoutHeader } from "@tsconline/shared";
+import XLSX from "xlsx";
+
+export async function parseExcelFile(filePath: string) {
+  const workbook = XLSX.readFile(filePath);
+  const sheetName: string = workbook.SheetNames[0] || "";
+  const sheet = workbook.Sheets[sheetName];
+  if (!sheet) return [];
+
+  const jsonData: string[][] = XLSX.utils.sheet_to_json(sheet, { header: 1 });
+
+  const dataWithoutHeader = jsonData.slice(1);
+
+  // if (dataWithoutHeader.length > 0) {
+  //   dataWithoutHeader[0] = dataWithoutHeader[0].slice(3);
+  // } else {
+  //   assertTimescaleDataWithoutHeader(dataWithoutHeader);
+  // }
+
+  if (!dataWithoutHeader || dataWithoutHeader.length === 0) {
+    assertTimescaleDataWithoutHeader(dataWithoutHeader);
+    return [];
+  }
```

hmmm this is where you would change it but that return will never get hit because that `assert` method throws an error if it's empty and if it reaches this point, it will always be empty. you probably should just throw an error then and not return and deprecate `assertTimescaleDataWithoutHeader`